tutorials/plot_background_statistics.py

```diff
@@ -0,0 +1,642 @@
+    t_lims = -stats.distributions.t.ppf(p_lims, n_subjects - 1)
+    p_lims = [-np.log10(0.05), -np.log10(0.001)]
+    # t plot
+    x, y = np.mgrid[0:width, 0:width]
+    surf = axes[0].plot_surface(x, y, np.reshape(t, (width, width)),
+                                rstride=1, cstride=1, linewidth=0,
+                                vmin=t_lims[0], vmax=t_lims[1], cmap='viridis')
+    axes[0].set(xticks=[], yticks=[], zticks=[],
+                xlim=[0, width - 1], ylim=[0, width - 1])
+    axes[0].view_init(30, 15)
+    cbar = plt.colorbar(ax=axes[0], shrink=0.75, orientation='horizontal',
+                        fraction=0.1, pad=0.025, mappable=surf)
+    cbar.set_ticks(t_lims)
+    cbar.set_ticklabels(['%0.1f' % t_lim for t_lim in t_lims])
+    cbar.set_label('t-value')
+    cbar.ax.get_xaxis().set_label_coords(0.5, -0.3)
+    if not show:
+        axes[0].set(title=title)
+        if mcc:
+            axes[0].title.set_weight('bold')
+    # p plot
+    use_p = -np.log10(np.reshape(np.maximum(p, 1e-5), (width, width)))
+    img = axes[1].imshow(use_p, cmap='inferno', vmin=p_lims[0], vmax=p_lims[1],
+                         interpolation='nearest')
+    axes[1].set(xticks=[], yticks=[])
+    cbar = plt.colorbar(ax=axes[1], shrink=0.75, orientation='horizontal',
+                        fraction=0.1, pad=0.025, mappable=img)
+    cbar.set_ticks(p_lims)
+    cbar.set_ticklabels(['%0.1f' % p_lim for p_lim in p_lims])
+    cbar.set_label('$-\log_{10}(p)$')
+    cbar.ax.get_xaxis().set_label_coords(0.5, -0.3)
+    if show:
+        text = fig.suptitle(title)
+        if mcc:
+            text.set_weight('bold')
+        plt.subplots_adjust(0, 0.05, 1, 0.9, wspace=0, hspace=0)
+        mne.viz.utils.plt_show()
+
+
+plot_t_p(ts[-1], ps[-1], titles[-1], mccs[-1])
+
+###############################################################################
+# "hat" variance adjustment
+# ~~~~~~~~~~~~~~~~~~~~~~~~~
+# The "hat" technique regularizes the variance values used in the t-test
+# calculation [1]_ to compensate for implausibly small variances.
+ts.append(ttest_1samp_no_p(X, sigma=sigma))
+ps.append(stats.distributions.t.sf(np.abs(ts[-1]), len(X) - 1) * 2)
+titles.append('$\mathrm{t_{hat}}$')
+mccs.append(False)
+plot_t_p(ts[-1], ps[-1], titles[-1], mccs[-1])
+
+###############################################################################
+# Non-parametric tests
+# ^^^^^^^^^^^^^^^^^^^^
+# Instead of assuming an underlying Gaussian distribution, we could instead
+# use a **non-parametric resampling** method. Under the null hypothesis,
+# we have the principle of **exchangeability**, which means that, if the null
+# is true, we should be able to exchange conditions and not change the
+# distribution of the test statistic.
+#
+# In the case of a 2-tailed paired t-test against 0 (or between two conditions
+# where you have already subtracted them), exchangeability means that we can
+# flip the signs of our data. Therefore, we can
```

Comment:
Remove line break?

Reply:
Doesn't matter much because it renders fine